src/GovUk.Education.ExploreEducationStatistics.Data.Model.Tests/Repository/FootnoteRepositoryTests.cs

```diff
@@ -2615,997 +953,491 @@ await using (var context = InMemoryStatisticsDbContext(contextId))
+                    .Generate(2))
+                .GenerateList();
+            
+            var (subjectWithFootnotes, subjectWithNoFootnotes) = GetSubjectsTuple2(releaseSubjects);
+            
+            var subjectWithFootnotesIndicatorGroup = _fixture.DefaultIndicatorGroup()
+                .WithSubject(subjectWithFootnotes)
+                .WithIndicators(_fixture.DefaultIndicator().Generate(1))
+                .Generate();
+        
+            var releaseFootnote = _fixture.DefaultReleaseFootnote()
+                .WithRelease(release)
+                .WithFootnote(_fixture
+                    .DefaultFootnote()
+                    .WithSubjects(ListOf(subjectWithFootnotes))
+                    .WithFilters(ListOf(subjectWithFootnotes.Filters[0]))
+                    .WithFilterGroups(ListOf(subjectWithFootnotes.Filters[1].FilterGroups[0]))
+                    .WithFilterItems(ListOf(subjectWithFootnotes.Filters[2].FilterGroups[0].FilterItems[0]))
+                    .WithIndicators(ListOf(subjectWithFootnotesIndicatorGroup.Indicators[0]))
+                    .Generate())
+                .Generate();
+            
             var contextId = Guid.NewGuid().ToString();
-
+        
             await using (var context = InMemoryStatisticsDbContext(contextId))
             {
-                await context.AddRangeAsync(release);
-                await context.AddRangeAsync(
-                    releaseSubject1,
-                    releaseSubject2,
-                    releaseSubject3,
-                    releaseSubject4,
-                    releaseSubject5,
-                    releaseSubjectWithNoFootnotes
-                );
-                await context.AddRangeAsync(footnote);
+                await context.Release.AddRangeAsync(release);
+                await context.ReleaseSubject.AddRangeAsync(releaseSubjects);
+                await context.ReleaseFootnote.AddRangeAsync(releaseFootnote);
                 await context.SaveChangesAsync();
             }
-
+        
             await using (var context = InMemoryStatisticsDbContext(contextId))
             {
                 var repository = BuildFootnoteRepository(context);
                 var results = await repository.GetSubjectsWithNoFootnotes(release.Id);
-
+        
                 Assert.Single(results);
-
-                Assert.Equal(releaseSubjectWithNoFootnotes.Subject.Id, results[0].Id);
+        
+                Assert.Equal(subjectWithNoFootnotes.Id, results[0].Id);
             }
         }
 
+        private static Tuple<Subject, Subject> GetSubjectsTuple2(List<ReleaseSubject> releaseSubjects)
+        {
+            return releaseSubjects.Select(rs => rs.Subject).ToTuple2();
+        }
+
+        private static Tuple<Subject, Subject, Subject> GetSubjectsTuple3(List<ReleaseSubject> releaseSubjects)
+        {
+            return releaseSubjects.Select(rs => rs.Subject).ToTuple3();
+        }
```

Comment:
Similar to `FootnoteServiceTests`, do we need these? Could just inline them?

Reply:
I'd keep them here personally, as they are used multiple times over, and it was getting repetitive copy and pasting it over and again in different test methods.